### PR TITLE
feat: [PIE-2639]: OPA Policy Step

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
##### Summary:

Custom Policy Step addition to pipeline
Execution Content view update

##### Jira Links:

https://harness.atlassian.net/browse/PIE-2639
https://harness.atlassian.net/browse/PIE-2895

##### Screenshots:

https://user-images.githubusercontent.com/96711906/156548480-bb4dd29a-7a3b-48ab-aaa6-353a0583dacc.mov

https://user-images.githubusercontent.com/96711906/156548671-dc059f29-10ec-4867-90f0-9648033eb908.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by AshwinBhatkal-Harness** [comment]
trigger jest

File: src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            <FailureInfo label={getString('error')}>
+              <Text key={errorMessage} color={statusColor}>
+                {errorMessage}
+              </Text>
+            </FailureInfo>
+          )}
+        </Layout.Vertical>
+      </Layout.Vertical>
+      <ExecutionStatusLabel
+        status={
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx:171]: I think there is utility function for this. You can check with Vivek for that.

File: src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                >
+                  {getString('pipeline.policyEvaluations.policySetName', { name })}
+                </Link>
+              </Text>
+              <Text width={'100px'} font={{ size: 'small' }}>
+                {scope}
+              </Text>
+              <ExecutionStatusLabel
+                status={
+                  status === EvaluationStatus.PASS
```
Comment (vkbansal) [inline src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx:97]: Looks like you are repeating this logic. Probably abstract into a separate function. Will also decrease cognitive complexity in sonar

File: src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                    ? 'Success'
+                    : status === EvaluationStatus.ERROR
+                    ? 'Errored'
+                    : 'InterventionWaiting'
+                }
+                label={
+                  status === EvaluationStatus.PASS
```
Comment (vkbansal) [inline src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx:104]: Looks like you are repeating this logic. Probably abstract into a separate function. Will also decrease cognitive complexity in sonar

File: src/modules/70-pipeline/components/execution/StepDetails/tabs/StepDetailsTab/StepDetailsTab.tsx [hunk 4]
```diff
@@ -48,7 +53,8 @@ export function StepDetailsTab(props: ExecutionStepDetailsTabProps): React.React
         </div>
       ) : null}
       <StepDetails step={step} />
-      <LogsContent mode="step-details" toConsoleView={logUrl} />
+      {shouldShowPolicyEvaluationContent && <PolicyEvaluationContent step={step} />}
```
Comment (vkbansal) [inline src/modules/70-pipeline/components/execution/StepDetails/tabs/StepDetailsTab/StepDetailsTab.tsx:56]: Finally, we remove this 🥇

File: src/modules/75-cd/components/PipelineSteps/PolicyStep/PolicySets/PolicySetModal/__tests__/__snapshots__/PolicySetModal.test.tsx.snap [hunk 5]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Test Policy Set Modal snapshot test and modal is navigable 1`] = `<div />`;
```
Comment (vkbansal) [inline src/modules/75-cd/components/PipelineSteps/PolicyStep/PolicySets/PolicySetModal/__tests__/__snapshots__/PolicySetModal.test.tsx.snap:3]: empty snapshot?

File: src/modules/70-pipeline/components/execution/StepDetails/tabs/StepDetailsTab/StepDetailsTab.tsx (hunk 4 above)
Comment (vkbansal) [inline src/modules/70-pipeline/components/execution/StepDetails/tabs/StepDetailsTab/StepDetailsTab.tsx:56]: This should be implemented using factory pattern

File: src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx:171]: Extracted it

File: src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx (hunk 2 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx:97]: Extracted it

File: src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx (hunk 3 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/70-pipeline/components/execution/StepDetails/common/ExecutionContent/PolicyEvaluationContent/PolicyEvaluationContent.tsx:104]: Extracted it

File: src/modules/70-pipeline/components/execution/StepDetails/tabs/StepDetailsTab/StepDetailsTab.tsx (hunk 4 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/70-pipeline/components/execution/StepDetails/tabs/StepDetailsTab/StepDetailsTab.tsx:56]: Done

**Comment by AshwinBhatkal-Harness** [comment]
retrigger jest

**Comment by AshwinBhatkal-Harness** [comment]
retrigger coverage

**Review by vkbansal — APPROVED** [review]

**Review by Tamaraius — APPROVED** [review]

**Pull request merged**